src/NonCumulative/Soundness/Properties/Inversion.agda

```diff
@@ -0,0 +1,160 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Level hiding (_⊔_)
+open import Axiom.Extensionality.Propositional
+
+-- Properties of the gluing models for terms and types
+module NonCumulative.Soundness.Properties.Inversion (fext : ∀ {ℓ₁ ℓ₂} → Extensionality ℓ₁ ℓ₂) where
+
+open import Lib
+open import Data.Nat
+open import Data.Nat.Properties as ℕₚ
+
+open import NonCumulative.Semantics.Properties.PER (fext)
+open import NonCumulative.Soundness.LogRel
+open import NonCumulative.Soundness.Properties.LogRel (fext)
+
+-- this set of lemmas re-tie the knots to change all the Univ and Glu to their standard forms
+-- i.e., 𝕌-wellfounded _ and Glu-wellfounded _
+
+®-Π-inv : ∀ {i j k} →
```

Comment:
why call them inversion?

Reply:
I'm bad at naming so just pick some names. maybe change to sth like ®-Π-𝕌

Reply:
yeah it's more like conversion, or packaging.